docs/stable/dev/building/python.md

```diff
@@ -5,17 +5,291 @@ redirect_from:
+
+```bash
+$ git clone https://github.com/duckdb/duckdb.git
+...
+$ cd duckdb
+```
+
+If you've _forked_ DuckDB you may run into trouble when building the Python package when you haven't pulled in the tags.
+
+```bash
+# Check your remotes
+git remote -v
+
+# If you don't see upstream	git@github.com:duckdb/duckdb.git, then add it
+git remote add upstream git@github.com:duckdb/duckdb.git
+
+# Now you can pull & push the tags
+git fetch --tags upstream
+git push --tags
+```
+
+### 2. Python Virtual Env
+
+For everything described here you will need a suitable Python installation. While you technically might be able to use your system Python, we **strongly** recommend you use a Python virtual environment. A virtual environment isolates dependencies and, depending on the tooling you use, gives you control over which Python interpreter you use. This way you don't pollute your system-wide Python with the different packages you need for your projects.
+
+While we use Python's built-in `venv` module in our examples below, and technically this might (or migth not!) work for you, we also **strongly** recommend use a tool like [astral uv](https://docs.astral.sh/uv/) (or Poetry, conda, etc) that allows you to manage _both_ Python interpreter versions and virtual environments.
+
+Create and activate a virtual env as follows:
+
+```bash
+# Create a virtual environment in the .venv folder (in the duckdb source root)
+$ python3 -m venv --prompt duckdb .venv
+
+# Activate the virtual env
+$ source .venv/bin/activate
+```
+
+Make sure you have a modern enough version of pip available in your virtual env:
+
+```bash
+# Print pip's help
+$ python3 -m pip install --upgrade pip
+```
+
+If that fails with `No module named pip` and you use `uv`, then run:
+
+```bash
+# Install pip
+$ uv pip install pip
+```
+
+## Building From Source
+
+Below are a number of options to build the python library from source, with or without debug symbols, and with a default or custom set of [extensions]({% link docs/stable/extensions/overview.md %}). Make sure to check out the [DuckDB build documentation]({% link docs/stable/dev/building/overview.md %}) if you run into trouble building the DuckDB main library.
+
+### Default release, debug build or cloud storage
+
+The following will build the package with the default set of extensions (json, parquet, icu and core_function).
+
+#### Release build
+
+```bash
+GEN=ninja BUILD_PYTHON=1 make release
+```
+
+#### Debug build
+
+```bash
+GEN=ninja BUILD_PYTHON=1 make debug
+```
+
+#### Cloud Storage
+
+You may need the package files to reside under the same prefix where the library is installed; e.g., when installing to cloud storage from a notebook.
+
+First, get the repository based version number and extract the source distribution.
+
+```bash
+python3 -m pip install build # required for PEP 517 compliant source dists
+cd tools/pythonpkg
+export SETUPTOOLS_SCM_PRETEND_VERSION=$(python3 -m setuptools_scm)
```

Comment:
I'm not entirely sure about this, as I haven't tested the use of `SETUPTOOLS_SCM_PRETEND_VERSION` myself.
`OVERRIDE_GIT_DESCRIBE` is more battle-tested I feel, but only by a little. The `setup.py` has both, and also `MAIN_BRANCH_VERSIONING` has impact on the generated version (v1.3.0 vs 1.2.3 for example).

Reply:
To explain this in a bit more detail:
when creating the sdist, we use the amalgamation to collect the files, then inside `src/function/table/version/pragma_version.cpp` we add defines for the:
- `DUCKDB_SOURCE_ID`
- `DUCKDB_VERSION`
- `DUCKDB_MAJOR_VERSION`
- `DUCKDB_MINOR_VERSION`
- `DUCKDB_PATCH_VERSION`

This is what's used to create the version of duckdb that gets installed.
Through the `use_scm_version` keyword argument in the `setup()` call we make sure the python package versioning matches this version.

I don't know how much of this we want to add to the docs here, but I figured I'd add some context around this

Reply:
This is a section I've copied over verbatim [from the README](https://github.com/duckdb/duckdb/pull/17483/files#diff-565fa0ef6a3eabd7c23791fb3cd6dc41c0b7505d9cabd48cbf7aad44496abab1L54). It's from a 5+ year old commit (by @arjenpdevries :-)) and hasn't really been touched since then, by the looks of it. I'm not sure what exactly this installation option is and whether it's still needed... I can also just delete it from the docs if it's no longer relevant?

Reply:
I deleted this section. It's not working anymore anyway. If we do at some point need instructions for this sort of install then we can revisit and add them.